### PR TITLE
PLATFORM 486 WIP

### PR DESCRIPTION
**Pull request by awalsh272**
Adding functions for restCall and mainRecords into the appropriate repo, with some changes from the original made. Also added utility functions that are used in the data_loading script. Envisioning having separate py scripts for each survey form

### CODE REVIEW

File: puente-data-exporter/lambdas/hello-world/restCall.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import requests
+import json
+from pandas import json_normalize
+
+import secretz
+
+def lambda_handler(event, context):
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/restCall.py]: This handler should replace the handler in hello-world>index.py

File: puente-data-exporter/lambdas/hello-world/restCall.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import requests
+import json
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/restCall.py]: For all of your imports that are not native to python like request, pandas etc. You should create a requirement.txt that contains the name of every library import. Can show you what i mean if unclear. File should be located in hello-world folder.

File: puente-data-exporter/lambdas/hello-world/utils.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
+import pandas as pd
+import numpy as np
+import re
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/utils.py]: these two should also be added to requirements.txt

File: puente-data-exporter/lambdas/hello-world/secretz.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+APP_ID = "vBdTHqQU31IyLW5uYRDIWb8Ew8zCZGBzMqChugjr"
+
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/secretz.py]: remove from PR

File: puente-data-exporter/lambdas/hello-world/restCall.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    #     # Send some context about this error to Lambda Logs
+    #     print(e)
+
+    #     raise e
+    # def restCall(specifier, url="https://parseapi.back4app.com/classes/"):
+    # """Only difference between calls is the specifier
+    #    This function makes a REST Get call to Back4App, returning a pandas df"""
+
+    url="https://parseapi.back4app.com/classes/"
+    specifier = "SurveyData"
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/restCall.py]: should try to read the specficer from the event. The request coming in should look like this:

{
   surveyingOrganization: "Puente",
   parseClass: "SurveyData"
}

So you would want to insert those parameters in for specifier and where "OWS" is. Let me know if that needs clearing up.

File: puente-data-exporter/lambdas/hello-world/restCall.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    combined_url = url + specifier
+
+    # REST stuff to make sure you get data
+    #HAD PROBLEMS WITH PAYLOAD SIZE, LOWERING LIMIT FIXED
+    #increase payload?
+    params = {
+      "order": "-updatedAt", "limit": 200000, "where":{
+          json.dumps({
+            "surveyingOrganization": {
+                "$in": ['OWS']
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/restCall.py]: surveyingOrganization request param instead of OWS

**Review by joseph-mccombs — CHANGES_REQUESTED** [review]
A few little things. god work!

File: puente-data-exporter/lambdas/hello-world/index.py [hunk 7]
```diff
@@ -1,9 +1,18 @@
...
-    'isBase64Encoded': False,
-    'statusCode': 200,
-    'headers': {},
-    'multiValueHeaders': {},
-    'body': 'Hello, Puente Team!'
-  }
+  survey_org = event["surveyingOrganization"]
+  specifier = event["specifier"]
+
+  if specifier == "SurveyData":
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/index.py]: Right idea here. For some of the checks we're going to be doing, I want to make the rest call standard prior to doing all of the work for each form like cleaning. I essentially want:

restCall(specifier, survey_org)

* we will eventually do some checks here but not currently

Then break it out into the other form specfic work.

File: puente-data-exporter/lambdas/hello-world/index.py [hunk 8]
```diff
@@ -1,9 +1,18 @@
...
+    response = mainRecords(survey_org)
+  elif specifier == "HistoryEnvironmentalHealth":
+    response = envHealth(survey_org)
+  elif specifier == "EvaluationMedical":
+    response = evalMedical(survey_org)
+  else:
+    print("Pick a valid specifier")
```
Comment (joseph-mccombs) [inline puente-data-exporter/lambdas/hello-world/index.py]: We do have vitals and customForms too :)

**Review by joseph-mccombs — CHANGES_REQUESTED** [review]

File: puente-data-exporter/lambdas/hello-world/restCall.py (hunk 1 above)
Comment (awalsh272) [reply puente-data-exporter/lambdas/hello-world/restCall.py]: index.py is the lambda, with fcn name "handler". Other scripts are called/referenced in index

File: puente-data-exporter/lambdas/hello-world/secretz.py (hunk 4 above)
Comment (awalsh272) [reply puente-data-exporter/lambdas/hello-world/secretz.py]: I think I did this with a new commit?

File: puente-data-exporter/lambdas/hello-world/restCall.py (hunk 5 above)
Comment (awalsh272) [reply puente-data-exporter/lambdas/hello-world/restCall.py]: Should be good now

File: puente-data-exporter/lambdas/hello-world/restCall.py (hunk 6 above)
Comment (awalsh272) [reply puente-data-exporter/lambdas/hello-world/restCall.py]: Should be good now

File: puente-data-exporter/lambdas/hello-world/utils.py (hunk 3 above)
Comment (awalsh272) [reply puente-data-exporter/lambdas/hello-world/utils.py]: added a requirements.txt

**Pull request closed without merging**